### ISSUE TITLE
Header with dot inside variable names make DataFrame not queryable

### ISSUE THREAD

**Issue by Krakitos** [databricks/spark-csv#120]
Consider the following CSV structure : session.connectionDuration, travel.nbVisitedMap, travel.nbMapMovement, travel.mapCycleFactor, network.meanDelay, network.stdDevDelay

I read the file using the following :

```java
DataFrame df = sqlContext.read()
                .format("com.databricks.spark.csv")
                .option("header", "true")
                .option("delimiter", ";")
                .option("inferSchema", "true")
                .option("mode", "FAILFAST") //RuntimeException if malformated line
                .load(learningPath);
```
When I ask to output the schema, all go fine : 

```java
System.out.println(Arrays.toString(df.schema().fieldNames()));

[session.connectionDuration, travel.nbVisitedMap, travel.nbMapMovement, travel.mapCycleFactor, network.meanDelay, network.stdDevDelay]
```

But when trying to manipulate the Dataframe, I got the following error : 

```java
org.apache.spark.sql.AnalysisException: Cannot resolve column name "session.connectionDuration" among (session.connectionDuration, travel.nbVisitedMap, travel.nbMapMovement, travel.mapCycleFactor, network.meanDelay, network.stdDevDelay);
```

**Comment by marmbrus** [comment]
Which version of Spark are you using?  You are maybe hitting: https://issues.apache.org/jira/browse/SPARK-5632

**Comment by Krakitos** [comment]
I'm using Spark 1.4.1

**Comment by schnee** [comment]
I get this same behavior on Spark 1.3.1

**Comment by barrybecker4** [comment]
Its unclear to me how we are supposed to work with datatset that have columns with dots (.) in them. When I try to do a groupBy on a dataFrame with such a dataset I get an error like:
org.apache.spark.sql.AnalysisException: Cannot resolve column name "unique. string" among (unique. string, string-normal, string/odd...)

Spark uses the dots to indicate subfields or something like that, but what if the dot is legitimately part of the column name?

**Comment by HyukjinKwon** [comment]
I think https://github.com/databricks/spark-csv/issues/203 is also related with this.

**Comment by HyukjinKwon** [comment]
For me I agree with @barrybecker4. If some characters are not allowed in the column names in DataFrame at Spark, I think users simply should not use them. This might be simply done with a transformation as the header is located at the first.

**Comment by HyukjinKwon** [comment]
I see special characters are supported in Spark.

**Comment by barrybecker4** [comment]
Using backquotes around the column names is an acceptable workaround for now.

**Comment by HyukjinKwon** [comment]
I see. Thanks. However, does that look an issue at Spark but at this library? I think we may close this.

**Comment by barrybecker4** [comment]
Yes, the issue I mentioned is potentially an issue for spark and not databricks.csv.
